src/index.js

```diff
@@ -29,34 +29,42 @@ class Anvil {
     this.limiter = new RateLimiter(this.requestLimit, this.requestLimitMS, true)
   }
 
-  fillPDF (pdfTemplateID, payload) {
-    return this.requestREST(`/api/v1/fill/${pdfTemplateID}.pdf`, {
-      method: 'POST',
-      json: payload,
-      encoding: null,
-      headers: { Authorization: this.authHeader },
-    })
+  fillPDF (pdfTemplateID, payload, clientOptions = {}) {
+    return this.requestREST(
+      `/api/v1/fill/${pdfTemplateID}.pdf`,
+      {
+        method: 'POST',
+        body: JSON.stringify(payload),
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: this.authHeader,
+        },
+      },
+      clientOptions,
+    )
   }
 
   // Private
 
-  async requestREST (url, options) {
-    const optionsWithURL = {
-      ...options,
-      url: this.url(url),
-    }
-
+  async requestREST (url, options, clientOptions = {}) {
     return this.throttle(async (retry) => {
-      const { response, data } = await this.requestPromise(optionsWithURL)
-      const statusCode = response.statusCode
+      const response = await this.request(url, options)
+      const statusCode = response.status
+
       if (statusCode === 429) {
-        return retry(getRetryMS(response.headers['retry-after']))
+        return retry(getRetryMS(response.headers.get('retry-after')))
       }
+
       if (statusCode >= 300) {
-        const isObject = data && data.constructor.name === 'Object'
-        if (isObject && data.errors) return { statusCode, ...data }
-        else if (isObject && data.message) return { statusCode, errors: [data] }
+        const json = await response.json()
+        const errors = json.errors || (json.message && [json.message])
+
+        return errors ? { statusCode, errors } : { statusCode, ...json }
       }
+
+      const { dataType } = clientOptions
+      const data = dataType === 'stream' ? response.body : await response.buffer()
```

Comment:
`response.body` is technically a [PassThrough](https://nodejs.org/api/stream.html#stream_class_stream_passthrough), but for all intents and purposes, it's a stream. Not sure how explicit we might want to be about this in our documentation.